src/ui/component/fileList/view.jsx

```diff
@@ -1,165 +1,70 @@
-                ? this.props.claimsById[fileInfo1.claim_id].height
-                : 0;
-              const height2 = this.props.claimsById[fileInfo2.claim_id]
-                ? this.props.claimsById[fileInfo2.claim_id].height
-                : 0;
-
-              if (height1 !== height2) {
-                // flipped because heigher block height is newer
-                return height2 - height1;
-              }
-
-              if (fileInfo1.absolute_channel_position && fileInfo2.absolute_channel_position) {
-                return fileInfo1.absolute_channel_position - fileInfo2.absolute_channel_position;
-              }
-
-              return 0;
-            })
-          : [...fileInfos].reverse(),
-      [SORT_OPTIONS.DATE_OLD]: fileInfos =>
-        this.props.sortByHeight
-          ? fileInfos.slice().sort((fileInfo1, fileInfo2) => {
-              const height1 = this.props.claimsById[fileInfo1.claim_id]
-                ? this.props.claimsById[fileInfo1.claim_id].height
-                : 999999;
-              const height2 = this.props.claimsById[fileInfo2.claim_id]
-                ? this.props.claimsById[fileInfo2.claim_id].height
-                : 999999;
-              if (height1 < height2) {
-                return -1;
-              } else if (height1 > height2) {
-                return 1;
-              }
-              return 0;
-            })
-          : fileInfos,
-      [SORT_OPTIONS.TITLE]: fileInfos =>
-        fileInfos.slice().sort((fileInfo1, fileInfo2) => {
-          const getFileTitle = fileInfo => {
-            const { value, name, claim_name: claimName } = fileInfo;
-            if (value) {
-              return value.title || claimName;
-            }
-
-            // Invalid claim
-            return '';
-          };
-          const title1 = getFileTitle(fileInfo1).toLowerCase();
-          const title2 = getFileTitle(fileInfo2).toLowerCase();
-          if (title1 < title2) {
-            return -1;
-          } else if (title1 > title2) {
-            return 1;
-          }
-          return 0;
-        }),
-      [SORT_OPTIONS.FILENAME]: fileInfos =>
-        fileInfos.slice().sort(({ file_name: fileName1 }, { file_name: fileName2 }) => {
-          const fileName1Lower = fileName1.toLowerCase();
-          const fileName2Lower = fileName2.toLowerCase();
-          if (fileName1Lower < fileName2Lower) {
-            return -1;
-          } else if (fileName2Lower > fileName1Lower) {
-            return 1;
-          }
-          return 0;
-        }),
-    };
-  }
-
-  getChannelSignature = (fileInfo: { pending: boolean } & FileListItem) => {
-    if (fileInfo.pending) {
-      return undefined;
-    }
+export default function FileList(props: Props) {
+  const { uris, header, sort, injectedItem, loading, persistedStorageKey, noHeader, slim, empty } = props;
+  const [currentSort, setCurrentSort] = usePersistedState(persistedStorageKey || 'file-list-global-sort', SORT_NEW);
```

Comment:
so if I pass nothing or false, it still persists in a global namespace? IMO make all persisting here explicit and do not persist if a value isn't passed